#! /bin/sh -e

# DP: profiling fix

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

Date: Mon, 25 Feb 2002 01:54:30 -0800
From: Gary Hade <garyhade@us.ibm.com>
To: Randolph Chung <randolph@tausq.org>
Cc: linux-ia64@linuxia64.org, gcc-bugs@gcc.gnu.org
Subject: Re: [Linux-ia64] gcc profiling broken on Linux/ia64?

Randolph,
I believe this problem is due to a bug in /usr/lib/gcrt1.o
that was fixed by recent glibc changes to csu/gmon-start.c
(revs 1.13 and 1.14).

Hope this helps.

Gary

--- csu/gmon-start.c.orig	Mon Feb 25 01:10:03 2002
+++ csu/gmon-start.c	Wed Feb 13 17:08:05 2002
@@ -1,5 +1,5 @@
 /* Code to enable profiling at program startup.
-   Copyright (C) 1995, 1996, 1997, 2000, 2001 Free Software Foundation, Inc.
+   Copyright (C) 1995,1996,1997,2000,2001,2002 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -63,7 +63,11 @@
 #endif
 
   /* Start keeping profiling records.  */
+#ifdef ENTRY_POINT_DECL
+  __monstartup ((u_long) ENTRY_POINT, (u_long) &etext);
+#else
   __monstartup ((u_long) &ENTRY_POINT, (u_long) &etext);
+#endif
 
   /* Call _mcleanup before exiting; it will write out gmon.out from the
      collected data.  */


